apis/v1alpha1/validation.go

```diff
@@ -33,11 +35,54 @@ func ValidateResourceMapping(m *ResourceMapping) (vErr error) {
 			vErr = errors.Join(vErr, fmt.Errorf("invalid owner: %w", err))
 		}
 	}
-	if m.Resource.Provider == "" {
-		vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
-	}
+	// if m.Resource.Provider == "" {
+	// 	vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
+	// }
 	if _, ok := m.Annotations.AsMap()[AnnotationKeyAssetInfo]; ok {
 		vErr = errors.Join(vErr, fmt.Errorf("reserved key is included: %s", AnnotationKeyAssetInfo))
 	}
+
+	if err := validateResource(m.Resource); err != nil {
+		vErr = errors.Join(vErr, fmt.Errorf("invalid Resourc: %w", err))
+	}
+
 	return
 }
+
+func validateResource(r *Resource) (vErr error) {
+	if r.Name == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("empty resource name"))
+	}
+
+	if r.Provider == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
+	}
+
+	if err := validateAndNormalizeSubscope(r); err != nil {
+		vErr = errors.Join(vErr, fmt.Errorf("invalid subscope: %w", err))
+	}
+
+	return
+}
+
+func validateAndNormalizeSubscope(r *Resource) error {
+	if r.Subscope == "" {
+		return nil
+	}
+
+	// normalize subscope string to only have lower cases
+	r.Subscope = strings.ToLower(r.Subscope)
```

Comment:
Do we need to normalize it or do we need to do any validation? My concern is that in BQ entry, we will record an reference to the github file and github commit. If we do any changes to the user's input and the bq entry doesn't match the source of truth in github repo, it will bring confusion.

Reply:
The normalization here only apply to the subscope string. I added the comment to subscope to let the user know the string will be normalized.

Reply:
By the way, I think normalize here doesn't make the subscope all lower case when it goes to bigquery. I doubt users won't be catch the normalize part of  proto description when they query bq. Can we change it to validation that if they have non-lower cases, just throw error?

Reply:
Updated.